src/client/server.ts

```diff
@@ -25,21 +125,46 @@ function makePohttpClientPlugin(
 
   server.post('/', async (request, reply) => {
     const message: Message = { headers: request.headers, body: request.body };
-    let events: CloudEventV1<unknown>;
+    const event = HTTP.toEvent(message) as CloudEventV1<unknown>;
+
+    let eventData: EventData;
     try {
-      events = HTTP.toEvent(message) as CloudEventV1<unknown>;
-    } catch {
-      await reply
-        .status(HTTP_STATUS_CODES.BAD_REQUEST)
-        .send({ type: PohttpClientProblemType.INVALID_EVENT });
-      return;
+      eventData = getMessageData(event);
+    } catch (err) {
+      server.log.info({ eventId: event.id }, (err as Error).message);
+      return reply.status(HTTP_STATUS_CODES.BAD_REQUEST).send();
+    }
+
+    const channel = await getChannel(eventData, server.activeEndpoint, server.log, server.mongoose);
+    if (channel === null) {
+      return reply.status(HTTP_STATUS_CODES.SERVICE_UNAVAILABLE).send();
     }
 
-    const event = events as CloudEvent;
+    const parcelSerialised = await channel.makeMessage(
+      new ServiceMessage(eventData.dataContentType, eventData.data),
+      Parcel,
+      {
+        ttl: eventData.ttl,
+        id: eventData.id,
+        creationDate: eventData.creationDate,
+      },
+    );
+
+    try {
+      await deliverParcel(channel.peer.internetAddress, parcelSerialised, { useTls: true });
+    } catch (err) {
+      if (err instanceof PoHTTPInvalidParcelError) {
+        server.log.info({ err }, 'Delivery failed due to server refusing parcel');
+      } else if (err instanceof PoHTTPClientBindingError) {
+        server.log.info({ err }, 'Delivery failed due to server binding');
+      } else {
+        server.log.info({ err }, 'Retry due to failed delivery');
```

Comment:
This is describing an outcome that we can't codify in the app -- [it's up to the operator to do that](https://docs.relaycorp.tech/awala-endpoint-internet/integration#dead-letter-channel).

Also, this should be a warning.

```suggestion
        request.log.warn({ err }, 'Failed to deliver parcel');
```